add new text on llm as rater

neilernst opened the pull request:
this PR replaces #30 which had weird recursion problems due to an earlier merge

sbaltes commented:
Awesome. Let's merge this before anything breaks again :-)

@ctreude can then check the live version and post his feedback here.

sbaltes commented:
> From our MSR paper, it might be useful to mention model-model agreement and model confidence as two ways to estimate how well LLMs perform the rating task.

Very good point! We can/should refer to inter-model agreement as mentioned in the "Report Suitable Baselines, Benchmarks, and Metrics" and "Use an Open LLM as a Baseline" guidelines.

neilernst commented:
Hi both, I've updated and created PR #37 to fix Christoph's changes. My comments in *italics* if needed

- [x] explicitly call out the rating of artifacts produced by the researchers themselves
- [x] explicitly point to the sections on "Report Model Version and Configuration" and "Report Prompts and their Development" to reinforce transparency in reporting. 
- [x] I wouldn’t call it the "conventional Cohen’s kappa measure" 
- [x] Would a paragraph on the extent to which rating categories affect LLM output be in scope? *not a paragraph but added as a comment that LLMs are not deterministic*
- [x] highlight scale
- [x] should LLMs be expected to reflect the majority opinion or act as independent raters? *added as a challenge*

Things not yet resolved:

- [ ] the process for comparing LLM and human output should be described precisely -- including how many items were compared, how coding guidelines evolved, agreement at different stages, the number of human raters and LLMs involved, how disagreements were resolved, etc.
- [ ]  mention model-model agreement and model confidence as two ways to estimate how well LLMs perform the rating task. 
   - [ ] refer to inter-model agreement as mentioned in the "Report Suitable Baselines, Benchmarks, and Metrics" and "Use an Open LLM as a Baseline" guidelines.

I am not sure if these changes make sense for 'study types'. To me, the point of this section is to explain that there is this type of study and some examples, alongside pros and cons (why one might do such a thing with an LLM). 

These comments feel more like guidelines, *how to use an LLM as a judge*. Which should totally be a guideline - based on your tutorial Christoph. But that feels quite distinct from the study type. Maybe a way out is to show how such a study would be best done, by referring to the guidelines. I can see the need for this type of writeup esp once the review is finished.

sbaltes commented:
Hi Neil,

Thanks a lot!
The current approach is that the guidelines point to the study types, but not the other way around. So what we could do is have a paragraph on LLM as a judge/rater in a suitable part of the guidelines? I guess "Report Suitable Baselines, Benchmarks, and Metrics" would be quite fitting. What do you think?

sbaltes commented:
Also: In terms of "LLM as judge" vs. "LLM as rater": I'd be fine with either. I chose "rater" because judge sounds like a binary decision (guilty/not guilty) and rater would allow us to include more complex ratings. What do you two think @ctreude @neilernst?

sbaltes commented:
> So what we could do is have a paragraph on LLM as a judge/rater in a suitable part of the guidelines? I guess "Report Suitable Baselines, Benchmarks, and Metrics" would be quite fitting. What do you think?

@neilernst Ping :-)
